src/snowcli/cli/nativeapp/manager.py

```diff
@@ -319,6 +319,17 @@ def _create_dev_app(self, diff: DiffResult) -> None:
 
             # Create an app using "loose files" / stage dev mode.
             log.info(f"Creating new application {self.app_name} in account.")
+
+            if self.app_role != self.package_role:
+                with self.use_role(new_role="accountadmin"):
+                    # Application owner role should also have create database permission granted to it.
+                    self._execute_queries(
+                        f"""
+                        grant create database on account to role {self.app_role};
+                        grant install, develop on application package {self.package_name} to role {self.app_role};
+                        """
+                    )
```

Comment:
I think we can just `use_role(package_role)` and assume their app role already has the required permissions.

We should not depend on `accountadmin` being available.

Reply:
That's fair. 
Though the user will then need to `grant create database on account to role {self.app_role};` themselves in preparation to use it in our cli. 
Pushing changes with this in mind.

Reply:
I think in conjunction with the external docs changes to point them to the regular Native Apps docs around privilege requirements we should be good.